app/src/main/java/a3/com/convo/activities/LoginActivity.java

```diff
@@ -223,7 +223,10 @@ public void onCompleted(JSONObject object, GraphResponse response) {
                             final String id = object.getString("id");
                             final String email = object.getString("email");
                             final String name = object.getString("name");
-                            final String profPicUrl = object.getJSONObject("picture").getJSONObject("data").optString("url");
+                            final JSONObject j = object.getJSONObject("picture");
+                            final JSONObject k = j.getJSONObject("data");
+                            final String profPicUrl = k.optString("url");
+//                            final String profPicUrl = object.getJSONObject("picture").getJSONObject("data").optString("picture"); // this doesn't work for some reason
```

Comment:
Did you try changing this to
`final String profPicUrl = object.getJSONObject("picture").getJSONObject("data").optString("url");`

Lets not name objects j and k. Instead name them `picture, data, profilePicUrl`

Also make sure we add null checks for all these objects

Reply:
We've fixed this since! I'm not sure why it was reverted when I pushed. String profUrl = page.getJSONObject("picture").getJSONObject("data").optString("url"); works, I'll change this back.